clients/src/main/java/org/apache/kafka/common/security/authenticator/SaslServerAuthenticator.java

```diff
@@ -104,14 +106,16 @@
     private NetworkReceive netInBuffer;
     private Send netOutBuffer;
 
-    public SaslServerAuthenticator(String node, JaasContext jaasContext, final Subject subject, KerberosShortNamer kerberosNameParser, String host, int maxReceiveSize, CredentialCache credentialCache) throws IOException {
+    public SaslServerAuthenticator(String node, JaasContext jaasContext, final Subject subject,
+                                   KerberosShortNamer kerberosNameParser, String host, int maxReceiveSize,
```

Comment:
Is it even worth being able to pass `maxReceiveSize`? Probably in tests, I guess. Maybe we should document in the constructor Javadoc that the effective `maxReceiveSize` is always less than `MAX_RECEIVE_SIZE`.